[Web] tsc failure with 1.17.0 and "strict": true in tsconfig.json

### Describe the issue

`tsc` prints the following when type-checking a project that depends on onnxruntime-web 1.17.0. This did not happen with 1.16.x.

```
app/components/App.ts:3:22 - error TS7016: Could not find a declaration file for module 'onnxruntime-web'. '[…]/node_modules/onnxruntime-web/dist/esm/ort.min.js' implicitly has an 'any' type.
  There are types at '[…]/node_modules/onnxruntime-web/types.d.ts', but this result could not be resolved when respecting package.json "exports". The 'onnxruntime-web' library may need to update its package.json or typings.

3 import * as ort from 'onnxruntime-web'
```

### To reproduce

```bash
npx create-next-app@14.1.0 --no-eslint --no-tailwind --ts --app --no-src-dir --import-alias '@/*' test
cd test
npm i onnxruntime-web
mkdir app/components
cat > app/components/App.ts <<EOF
'use client'

import * as ort from 'onnxruntime-web'

export default function App() {
  console.log(ort.Tensor)
}
EOF
npx tsc
```

Make sure that `tsconfig.json` has `"strict": true`. If you change it so that it has `"strict": false` instead, the error goes away.

### Urgency

This prevents using `onnxruntime-web` in default NextJS apps.

### ONNX Runtime Installation

Released Package

### ONNX Runtime Version or Commit ID

1.17.0

### Execution Provider

'wasm'/'cpu' (WebAssembly CPU), 'webgpu' (WebGPU)

--- a/js/web/package.json
+++ b/js/web/package.json
@@ -69,11 +69,14 @@
   "exports": {
     ".": {
       "node": "./dist/ort.node.min.js",
+      "types": "./types.d.ts",
       "default": {
         "import": "./dist/esm/ort.min.js",
         "require": "./dist/cjs/ort.min.js",
+        "types": "./types.d.ts",
         "default": {
           "development": "./dist/ort.js",
+          "types": "./types.d.ts",
           "default": "./dist/ort.min.js"
         }
       }
@@ -81,34 +84,41 @@
     "./experimental": {
       "import": "./dist/esm/ort.all.min.js",
       "require": "./dist/cjs/ort.all.min.js",
+      "types": "./types.d.ts",
       "default": {
         "development": "./dist/ort.all.js",
+        "types": "./types.d.ts",
         "default": "./dist/ort.all.min.js"
       }
     },
     "./wasm": {
       "import": "./dist/esm/ort.wasm.min.js",
       "require": "./dist/cjs/ort.wasm.min.js",
+      "types": "./types.d.ts",
       "default": "./dist/ort.wasm.min.js"
     },
     "./wasm-core": {
       "import": "./dist/esm/ort.wasm-core.min.js",
       "require": "./dist/cjs/ort.wasm-core.min.js",
+      "types": "./types.d.ts",
       "default": "./dist/ort.wasm-core.min.js"
     },
     "./webgl": {
       "import": "./dist/esm/ort.webgl.min.js",
       "require": "./dist/cjs/ort.webgl.min.js",
+      "types": "./types.d.ts",
       "default": "./dist/ort.webgl.min.js"
     },
     "./webgpu": {
       "import": "./dist/esm/ort.webgpu.min.js",
       "require": "./dist/cjs/ort.webgpu.min.js",
+      "types": "./types.d.ts",
       "default": "./dist/ort.webgpu.min.js"
     },
     "./training": {
       "import": "./dist/esm/ort.training.wasm.min.js",
       "require": "./dist/cjs/ort.training.wasm.min.js",
+      "types": "./types.d.ts",
       "default": "./dist/ort.training.wasm.min.js"
     }
   },

